feat(ui-rest): Provided option in attachment usage to include/exclude concluded licenses during LicenseInfo Generation

JaideepPalit opened the pull request:
> Provided option in attachment usage to include/exclude concluded licenses during LicenseInfo Generation
> * Which issue is this pull request belonging to and how is it solving it? (*#1074*)
> * Did you add or update any new dependencies that are required for your change?

### How To Test? - Refer #1074 
> - Change Attachment Usage with different file types from Attachment Tab and From LicenseInfo/Report Generation Page.
      - Verify Attachement Usage with proper info are getting created and deleted from DB based on selection. 
> - Old Attachement Usages should function normally - by default conclusions are included.
> - Download LicenseInfo and Report in different format with/without conclusions from UI and Rest

> Have you implemented any additional tests? - No

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>

smrutis1 commented:
Few Remarks:

1) Created a new project and added a release having a spdx file, By default the conclusion checkbox is unselected and disabled. 

![image](https://user-images.githubusercontent.com/46560890/103057712-43e73880-45c6-11eb-83ae-eca165eadcdd.png)

2) Saved the usage by selecting both conclusions and attachment selection and refreshed the license selection page, how ever none of the checkboxes are selected in license selection page.

Attachment Usage Page:

![image](https://user-images.githubusercontent.com/46560890/103058389-05eb1400-45c8-11eb-9d15-d392f1c0f9a5.png)
License selection page:

![image](https://user-images.githubusercontent.com/46560890/103058547-73974000-45c8-11eb-9b40-6d9e02a5d982.png)

3) Tested attachment usage only with CLI file, selected both the files as per the screenshot and refreshed the license selection page, how ever the same is not reflecting. 
Attachment Usage Page:

![image](https://user-images.githubusercontent.com/46560890/103059478-413b1200-45cb-11eb-9ad7-80305ff46d37.png)

License Selection Page:
![image](https://user-images.githubusercontent.com/46560890/103059553-70518380-45cb-11eb-9439-d02c65205d24.png)

4) Seems like get attachment usage returns incorrect data on `includeConcludedLicense` field
![image](https://user-images.githubusercontent.com/46560890/103065483-aba87e00-45dc-11eb-9651-be3743a52f97.png)

5) Could the `Conclusions` column be hidden from UI for CLI files, if not applicable?

JaideepPalit commented:
Thanks @smrutis1 for review and test.
Added fix for all remarks mentioned above.

smrutis1 commented on backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java:
```diff
@@ -452,8 +463,8 @@ public String getDefaultObligationsText() {
 
-    protected Map<Release, Set<String>> mapKeysToReleases(Map<String, Set<String>> releaseIdsToAttachmentIds, User user) throws TException {
-        Map<Release, Set<String>> result = Maps.newHashMap();
+    protected Map<Release, Map<String,Boolean>> mapKeysToReleases(Map<String, Map<String,Boolean>> releaseIdsToAttachmentIds, User user) throws TException {
```
indentation

smrutis1 reviewed (changes requested):
One observation:

- Created a new project and added a release having an accepted CLI/spdx file, in the license info generation page the file is not selected. Expected behavior:- It should be selected by default.

JaideepPalit commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:
```diff
@@ -543,15 +543,18 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
         LicenseNameWithText licenseWithText = null;
         try {
             Release release = componentClient.getReleaseById(releaseId, user);
+            // need to revisit
```
Removed

JaideepPalit commented:
> * Created a new project and added a release having an accepted CLI/spdx file, in the license info generation page the file is not selected. Expected behavior:- It should be selected by default.
  - Fixed

smrutis1 reviewed (approved):
Code looks good to me. Tested successfully the attachment usage functionality.
